packages/fetch/src/types.ts

```diff
@@ -16,182 +16,177 @@
-  readonly bodyUsed: boolean;
-  arrayBuffer(): Promise<ArrayBuffer>;
-  blob(): Promise<unknown>;
-  json<ResponseBody = unknown>(): Promise<ResponseBody>;
-  text(): Promise<string>;
-}
-
-export interface FetchHeaders {
-  append(name: string, value: string): void;
-  delete(name: string): void;
-  get(name: string): string | null;
-  has(name: string): boolean;
-  set(name: string, value: string): void;
-  forEach(callback: (value: string, name: string) => void): void;
-}
-
-/** This Fetch API interface represents a resource request. */
-export interface FetchRequest extends FetchBody {
-  /**
-   * Returns a Headers object consisting of the headers associated with request. Note that headers added in the network layer by the user agent will not be accounted for in this object, e.g., the "Host" header.
-   */
-  readonly headers: FetchHeaders;
-  /**
-   * Returns a boolean indicating whether or not request can outlive the global in which it was created.
-   */
-  readonly keepalive: boolean;
-  /**
-   * Returns request's HTTP method, which is "GET" by default.
-   */
-  readonly method: string;
-  /**
-   * Returns the signal associated with request, which is an AbortSignal object indicating whether or not request has been aborted, and its abort event handler.
-   */
-  readonly signal: AbortSignal;
-  /**
-   * Returns the URL of request as a string.
-   */
-  readonly url: string;
-  clone(): FetchRequest;
-}
-
-export interface FetchResponse extends FetchBody {
-  readonly headers: FetchHeaders;
-  readonly ok: boolean;
-  readonly redirected: boolean;
-  readonly status: number;
-  readonly statusText: string;
-  readonly type: unknown;
-  readonly url: string;
-  clone(): FetchResponse;
-}
-
-export interface FetchRequestInit<RequestBody = unknown> {
+}; //  & AsyncIterable<Uint8Array>;
+
+export declare function fetch<
+  PlatformRequestBody = RequestBody,
+  PlatformHeaders extends Headers = Headers,
+  PlatformAbortSignal extends AbortSignal = AbortSignal,
+  PlatformResponseBody = ReadableStream | null,
+>(
+  input: string,
+  init?: RequestInit<PlatformRequestBody, PlatformHeaders, PlatformAbortSignal>,
+): Promise<Response<PlatformHeaders, PlatformResponseBody>>;
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export type AnyFetch = typeof fetch<any, any, any, any>;
+
+export interface RequestInit<
+  PlatformRequestBody = RequestBody,
+  PlatformHeaders = Headers,
+  PlatformAbortSignal = AbortSignal,
+> {
   /**
    * A BodyInit object or null to set request's body.
    */
-  body?: RequestBody;
+  body?: PlatformRequestBody;
   /**
    * A string indicating whether credentials will be sent with the request always, never, or only when sent to a same-origin URL. Sets request's credentials.
    */
-  credentials?: FetchRequestCredentials;
+  credentials?: RequestCredentials;
   /**
    * A Headers object, an object literal, or an array of two-item arrays to set request's headers.
```

Comment:
Similar comment as I wrote above. The docs for this type also refers to the previously allowed `string[][]`.
```suggestion
   * The request's headers.
```